hydromt/io/readers.py

```diff
@@ -575,15 +582,15 @@ def open_timeseries_from_table(fn, name=None, index_dim="index", **kwargs):
 
 
 def open_vector(
-    fn,
-    driver=None,
-    crs=None,
-    dst_crs=None,
-    bbox=None,
-    geom=None,
-    assert_gtype=None,
-    predicate="intersects",
-    mode="r",
+    fn: StrPath,
+    driver: Optional[OPEN_VECTOR_DRIVER] = None,
+    crs: Optional[CRS] = None,
+    dst_crs: Optional[CRS] = None,
+    bbox: Optional[Tuple[float]] = None,
+    geom: Optional[Union[gpd.GeoDataFrame, gpd.GeoSeries]] = None,
+    assert_gtype: Optional[Union[Point, LineString, Polygon]] = None,
+    predicate: OPEN_VECTOR_PREDICATE = "intersects",
+    logger: Logger = logger,
```

Comment:
Since #1011 we don't pass around loggers anymore, see migration guide for updates.

Reply:
I removed it